lib/cfg.c

```diff
@@ -764,56 +724,21 @@ void
 cfg_persist_config_add(GlobalConfig *cfg, const gchar *name, gpointer value, GDestroyNotify destroy,
                        gboolean force)
 {
-  PersistConfigEntry *p;
-
-  if (cfg->persist && value)
+  if (!cfg->persist && value)
     {
-      if (g_hash_table_lookup(cfg->persist->keys, name))
-        {
-          if (!force)
-            {
-              msg_error("Internal error, duplicate configuration elements refer to the same persistent config",
-                        evt_tag_str("name", name));
-              if (destroy)
-                destroy(value);
-              return;
-            }
-        }
-
-      p = g_new0(PersistConfigEntry, 1);
-
-      p->value = value;
-      p->destroy = destroy;
-      g_hash_table_insert(cfg->persist->keys, g_strdup(name), p);
+      if (destroy)
+        destroy(value);
       return;
     }
-  else if (destroy && value)
-    {
-      destroy(value);
-    }
-  return;
+  persist_config_add(cfg->persist, name, value, destroy, force);
```

Comment:
I think this changes functionality.
We should call `persist_config_add()` only if we have both persist and a value.
We we shouldn't store NULL values or when both value and persist is NULL.

Reply:
I don't think we would ever attempt to store a null value, but if we ever do it doesn't matter much as it fetch will return null in both cases. But you are right I'm fixing this

Reply:
I have now added to drop NULL explicitly in the cfg.c layer so this should be fixed.